kedro/contrib/io/azure/csv_blob.py

```diff
@@ -107,30 +111,60 @@ def __init__(
                 Here you can find all available arguments:
                 https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_csv.html
                 All defaults are preserved, but "index", which is set to False.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
 
         """
-        self._filepath = filepath
         self._container_name = container_name
         self._credentials = credentials if credentials else {}
```

Comment:
```suggestion
         self._credentials = copy.deepcopy(credentials) or {}
```

Reply:
I know this is not the change in this PR, but we want to detach the mutational change of credentials dict after initialising this class (same reasoning for using deepcopy to load args and save args). The same change applies to `blob_to_text_args` and `blob_from_text_args`

Reply:
And argument of `credentials` to be `credentials: Optional[Dict[str, Any]] = None`

Reply:
Makes sense, thanks! Should be done now. :)